azkaban-common/src/main/java/azkaban/event/EventData.java

```diff
@@ -0,0 +1,34 @@
+package azkaban.event;
+
+import azkaban.executor.Status;
+
+/**
+ * Carries a snapshot of the status data, suitable for asynchronous message passing.
+ */
+public class EventData {
+
+  private Status status;
+  private String nestedId;
+
+  public static EventData create(Status status) {
+    return new EventData(status);
+  }
+
+  private EventData(Status status) {
+    this.status = status;
+  }
+
+  public Status getStatus() {
+    return status;
+  }
+
+  public String getNestedId() {
+    return nestedId;
+  }
+
+  public EventData withNestedId(String nestedId) {
```

Comment:
No use case other than making it easy to see that "it's the nestedId that is being set here". I can change to have another `create` method and make both fields `final`.